src/sync.h

```diff
@@ -75,13 +68,16 @@ namespace librealsense
     };
     //sync_lock::ref = 0;
 
+    class synthetic_source_interface;
+
     struct syncronization_environment
     {
         synthetic_source_interface* source;
         //sync_lock& lock_ref;
         single_consumer_queue<frame_holder>& matches;
     };
 
+    typedef int stream_id;
```

Comment:
please remove this `typedef`, or prefer usage of suggestions:
1. It is defined in multiple locations - prefer usage of `using stream_id = int;`
2. It does not really represent a stream's id since the id is both the type and the index - prefer the name `stream_index`